src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/controller/GeneralPartnerControllerValidationTest.java

```diff
@@ -0,0 +1,189 @@
+            "The quick brown fox jumps over the lazy dog " +
+            "The quick brown fox jumps over the lazy dog\",\n" +
+            "      \"surname\": \"Bloggs\",\n" +
+            "      \"date_of_birth\": \"2001-01-01\",\n" +
+            "      \"nationality1\": \"British\",\n" +
+            "      \"nationality2\": \"\",\n" +
+            "      \"kind\": \"kind\",\n" +
+            "      \"etag\": \"tag\",\n" +
+            "      \"general_partner_type\": \"Person\"\n" +
+            "    }\n" +
+            "}";
+
+    private HttpHeaders httpHeaders;
+    private Transaction transaction;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    GeneralPartnerController generalPartnerController;
+
+    @MockBean
+    GeneralPartnerService generalPartnerService;
+
+    @MockBean
+    private TransactionInterceptor transactionInterceptor;
+
+    @BeforeEach
+    void setUp() {
+        httpHeaders = new HttpHeaders();
+        httpHeaders.add("ERIC-Access-Token", "passthrough");
+        httpHeaders.add("X-Request-Id", "123");
+        httpHeaders.add("ERIC-Identity", "123");
+
+        transaction = new Transaction();
+
+        this.mockMvc = MockMvcBuilders.standaloneSetup(generalPartnerController)
+                .setControllerAdvice(new GlobalExceptionHandler())
+                .build();
+    }
+
+    @Test
+    void shouldReturn201()  throws Exception {
+        GeneralPartnerDto dto = createDto();
+        dto.getData().setSurname("Bloggs");
+        String body = objectMapper.writeValueAsString(dto);
+
+        mockMvc.perform(post(GeneralPartnerControllerValidationTest.postUrl)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .characterEncoding("utf-8")
+                        .headers(httpHeaders)
+                        .requestAttr("transaction", transaction)
+                        .content(body))
+                .andExpect(status().isCreated());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            JSON_WITH_BELOW_MIN_FORENAME,
+            JSON_WITH_ABOVE_MAX_FORENAME,
+            JSON_WITH_BELOW_MIN_SURNAME,
+            JSON_WITH_ABOVE_MAX_SURNAME,
+            JSON_WITH_ABOVE_MAX_FORMERNAME
+    })
+    void shouldReturn400(String body)  throws Exception {
+        mockMvc.perform(post(GeneralPartnerControllerValidationTest.postUrl)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .characterEncoding("utf-8")
+                        .headers(httpHeaders)
+                        .requestAttr("transaction", transaction)
+                        .content(body))
+                .andExpect(status().isBadRequest());
+    }
+
+    private static String createLongString() {
+        StringBuilder sb = new StringBuilder();
+        for(int i = 0; i < 161; i++) {
+            sb.append("a");
+        }
+        return sb.toString();
```

Comment:
Could use the repeat function instead e.g.  `return "a".repeat(161);`

Reply:
Changed.